src/FileUpload.tsx

```diff
@@ -0,0 +1,31 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import React, { useState } from 'react';
+import axios, { AxiosResponse } from 'axios';
+
+const FileUpload = (url: string) => {
+  const [selectedFile, setSelectedFile] = useState<File | null>(null);
+  const [response, setResponse] = useState<AxiosResponse | null>(null);
+
+  const handleFileUpload = async (evt: React.ChangeEvent<HTMLInputElement>) => {
+    const files = evt.target.files;
+    if (files && files[0]) {
+      setSelectedFile(files[0]);
+    }
+
+    const formData = new FormData();
+    try {
+      if (selectedFile) {
```

Comment:
I think the logic here is making the button not work as intended. If you add a console log just before the if block it will show that `selectedFile` is null.

Reply:
Fixed